packages/x-components/src/components/__tests__/x-component-utils.spec.ts

```diff
@@ -1,38 +1,30 @@
-import { mount } from '@vue/test-utils';
-import Vue, { CreateElement } from 'vue';
-import {
-  getXComponentXModuleName,
-  isXComponent,
-  setXComponentXModuleName
-} from '../x-component.utils';
+import { mount, Wrapper } from '@vue/test-utils';
+import Vue from 'vue';
+import { XModuleName } from '../../x-modules/x-modules.types';
+import { getXComponentXModuleName, isXComponent } from '../x-component.utils';
 
 describe('testing the x-component utils', () => {
-  const testComponent = {
-    render(h: CreateElement) {
-      return h();
-    }
-  };
-  let testXComponentWrapper: Vue;
-
-  beforeEach(() => {
-    testXComponentWrapper = mount(testComponent) as unknown as Vue;
-  });
-
+  function renderComponent(xModule?: XModuleName): Wrapper<Vue> {
+    return mount({
+      xModule,
+      render(h) {
+        return h();
+      }
+    });
+  }
   it('sets and gets as XComponent name the passed name parameter to the passed component', () => {
-    setXComponentXModuleName(testXComponentWrapper, 'searchBox');
-    expect(getXComponentXModuleName(testXComponentWrapper)).toBe('searchBox');
+    expect(getXComponentXModuleName(renderComponent('searchBox').vm)).toBe('searchBox');
   });
 
   it('returns null if the passed component is not an XComponent', () => {
-    expect(getXComponentXModuleName(testXComponentWrapper)).toBeNull();
+    expect(getXComponentXModuleName(renderComponent().vm)).toBeNull();
```

Comment:
Now the component stores the module name in its options object. There is no need to do that in the `beforeCreate` hook like previously.